projects/fusion-ui/components/file-drag-and-drop/file-drag-and-drop.directive.ts

```diff
@@ -0,0 +1,145 @@
+import {Directive, ElementRef, EventEmitter, Input, Output, OnInit, AfterViewInit, OnDestroy, Renderer2} from '@angular/core';
+import {isNullOrUndefined} from '@ironsource/fusion-ui/utils';
+import {fromEvent, Subject} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+
+/**
+ * Directive 'fusionFileDragAndDrop' for file selection by file select dialog
+ * or by files drag and drop to the host area
+ */
+@Directive({
+    selector: '[fusionFileDragAndDrop]'
+})
+export class FileDragAndDropDirective implements OnInit, AfterViewInit, OnDestroy {
+    /**
+     * element ID for initial file selection dialog by click.
+     * if not provided used click on host
+     */
+    @Input() elementIdForClick: string;
+    //
+    /**
+     * for disabling file selection
+     */
+    @Input() set fileDragAndDropDisabled(value: boolean) {
```

Comment:
change the name to disabled and add a class when in disabled mode `fusion-file-drag-drop-disabled`

Reply:
Ok. just with `fu-file-drag-drop-disabled`